stripe/src/main/java/com/stripe/android/StripeSuspend.kt

```diff
@@ -0,0 +1,682 @@
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.retrieveSetupIntent(
+    clientSecret: String,
+    stripeAccountId: String? = this.stripeAccountId
+) = runOnIOAndThrowIfNull {
+    stripeRepository.retrieveSetupIntent(
+        clientSecret,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId
+        )
+    )
+}
+
+/**
+ * Retrieve a [Source] from a coroutine.
+ *
+ * See [Retrieve a source](https://stripe.com/docs/api/sources/retrieve).
+ * `GET /v1/sources/:id`
+ *
+ * @param sourceId the [Source.id] field of the desired Source object
+ * @param clientSecret the [Source.clientSecret] field of the desired Source object
+ * @param stripeAccountId Optional, the Connect account to associate with this request.
+ * By default, will use the Connect account that was used to instantiate the `Stripe` object, if specified.
+ *
+ * @return a [Source] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.retrieveSource(
+    @Size(min = 1) sourceId: String,
+    @Size(min = 1) clientSecret: String,
+    stripeAccountId: String? = this.stripeAccountId
+) = runOnIOAndThrowIfNull {
+    stripeRepository.retrieveSource(
+        sourceId,
+        clientSecret,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [SetupIntent] object.
+ *
+ * See [Confirm a SetupIntent](https://stripe.com/docs/api/setup_intents/confirm).
+ * `POST /v1/setup_intents/:id/confirm`
+ *
+ * @param confirmSetupIntentParams a set of params with which to confirm the Setup Intent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [SetupIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmSetupIntentSuspend(
+    confirmSetupIntentParams: ConfirmSetupIntentParams,
+    idempotencyKey: String? = null
+) = runOnIOAndThrowIfNull {
```

Comment:
Why add `Suspend` to the method name here?

Reply:
`Stripe#confirmSetupIntent` actually brings up an activity for possible authentication, I think this function is more analogous to `Stripe#confirmSetupIntentSynchronous`, should we just keep it `Stripe#confirmSetupIntent` ?

Reply:
Renamed to `Stripe#confirmSetupIntent` after offline sync